engine/crates/engine-v2/src/engine/rate_limit.rs

```diff
@@ -0,0 +1,92 @@
+use std::collections::HashMap;
+use std::net::IpAddr;
+use std::num::NonZeroU32;
+
+use futures_util::future::BoxFuture;
+use futures_util::FutureExt;
+use governor::Quota;
+use http::{HeaderName, HeaderValue};
+use serde_json::Value;
+
+use config::latest::{Config, GLOBAL_RATE_LIMITER};
+use runtime::rate_limiting::{Error, RateLimiter, RateLimiterContext};
+
+pub struct EngineRateLimitContext<'a>(pub &'a str);
+
+impl RateLimiterContext for EngineRateLimitContext<'_> {
+    fn header(&self, _name: HeaderName) -> Option<&HeaderValue> {
+        None
+    }
+
+    fn graphql_operation_name(&self) -> Option<&str> {
+        None
+    }
+
+    fn ip(&self) -> Option<IpAddr> {
+        None
+    }
+
+    fn jwt_claim(&self, _key: &str) -> Option<&Value> {
+        None
+    }
+
+    fn key(&self) -> Option<&str> {
+        Some(self.0)
+    }
+}
+
+#[derive(Default)]
+pub struct InMemoryRateLimiter {
```

Comment:
Should be in `runtime-local`

Reply:
the interface probably needs to be adjusted, but the implementation (`governor`) can't be in engine-v2